src/azure-cli/azure/cli/command_modules/role/_validators.py

```diff
@@ -13,25 +13,31 @@
 VARIANT_GROUP_ID_ARGS = ['object_id', 'group_id', 'group_object_id']
 
 
+def _validate_group(namespace, attr, value, group_filter, has_next_filter):
```

Comment:
Personally, I think the usage of `has_next_filter` lowers the cohesion of this function. It makes `_validate_group` rely on the external logic and dependent between executions. I prefer not to exact the logic to `_validate_group` but put it in the original `validate_group` instead.

Reply:
I'm also hesitated at it. @qianwens , want to hear your opinions.

Reply:
I think it is correct to have _validate_group function or the handle exception logic in validate_group looks complicated. But for _validate_group, I agree that we should not raise different kind of error message based on the has_next_filter paramete. Currently _validate_group 's return logic is not predictable, sometime it return false, sometime it raise error, could you please refine the logic here, or you can replace this functio with _get_group_count to just return the count and let the validate_group to raise the exception

---
In reply to: [387564846](https://github.com/Azure/azure-cli/pull/12426#discussion_r387564846) [](ancestors = 387564846)

Reply:
updated